src/python/test/test_dataset.py

```diff
@@ -11,18 +11,10 @@
     import z5py
 
 
-class TestDataset(unittest.TestCase):
-
-    def setUp(self):
+class TestDatasetMixin(object):
```

Comment:
Is multiple inheritance really necessary here?
As far as I can see having `TestDatasetBase(unittest.TestCase)` and `TestZarr/N5Dataset(TestDatasetBase)`
should work as well.
Just wondering if there is any significant difference, otherwise this is a matter of taste and I am ok with the current 
implementation.
Refactoring the tests  was necessary indeed.

Reply:
The problem is that if the base class inherits from TestCase, its tests will be run, and obviously it doesn't have a concrete file type to test against. `unittest` even picks up abstract TestCase subclasses (although the mixin here should be abstract, for clarity, even though it's a bit of a kludge to make it work on both py2 and 3). 

The other way to do it would be to have, say, the TestZarrDataset as the parent and have the N5 version inherit from that, but that has implications about precedence.

It looked neater before I hit an MRO issue and had to call `mixinSetUp` etc. manually rather than just defining the setUp in the mixin. I might see if the tests still work inheriting from the mixin first.

Reply:
I see the issue. I just remember that I solved this in another project by moving the test base class
to another module to not interfere with `unittest` and `unittest discover` picking up on it.
Anyway, I am fine with the mixin pattern and adopting it for other tests is a good idea.